webapp/src/views/projects/translations/context/TranslationsContext.tsx

```diff
@@ -1,4 +1,4 @@
-import React, { useState, useRef } from 'react';
+import React, { useState, useRef, useEffect, useMemo } from 'react';
```

Comment:
I am not sure if this way of state management is really that great, since it tents to create large files like this, which is also hard to split. To smaller peaces.

Reply:
Refactored